extract-features.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/python
```

Comment:
Is the idea here that one would do something like this?

```
$ python3 extract-features.py >> Cargo.toml
```

Reply:
Yeah, that's exactly what I did here.  Updating/evolving it over time is a bit manual and clunky

Reply:
We may be able to use `println!` in build.rs to build the set of features dynamically http://doc.crates.io/build-script.html See `rustc-cfg=FEATURE`